### PR TITLE
feat: Updates in CI trigger modal 

### PR DESCRIPTION
**Pull request by sohel-rp**
# Description

These changes contain enhancements in the CI trigger modal from trigger view flows/interactions like fetching from the repo, searching of hash, and some other interactions.

Fixes [AB#989](https://dev.azure.com/DevtronLabs/Devtron/_workitems/edit/989)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

All tests have been performed manually by visiting the trigger view (Build & Deploy) and performing actions like re-fetching, searching of hash, etc.

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/app/details/triggerView/EmptyStateCIMaterial.tsx [hunk 1]
```diff
@@ -71,7 +71,7 @@ export class EmptyStateCIMaterial extends Component<EmptyStateCIMaterialProps> {
     let { title, subtitle, img, cta } = this.getData();
     if (this.props.isMaterialLoading) {
       return <EmptyState>
-        <EmptyState.Loading text={"Fetching repository. This might take few minutes"} />
+        <EmptyState.Loading text="Fetching... This might take few minutes" />
```
Comment (vivek-devtron) [inline src/components/app/details/triggerView/EmptyStateCIMaterial.tsx:74]: Please move messaging to const

File: src/components/app/details/triggerView/MaterialSource.tsx [hunk 2]
```diff
@@ -69,18 +69,19 @@ export class MaterialSource extends Component<MaterialSourceProps> {
         return (
             <>
                 {this.props.material.map((material, index) => {
-                    let selectedClass = material.isSelected ? 'material-selected' : ''
                     return (
                         <div
                             key={index}
-                            className={`material-list__item ${selectedClass}`}
+                            className={`material-list__item ${material.isSelected ? 'material-selected' : ''}`}
                             onClick={(e) => {
```
Comment (vivek-devtron) [inline src/components/app/details/triggerView/MaterialSource.tsx:76]: Please extract this out

File: src/components/app/details/triggerView/ciMaterial.tsx [hunk 3]
```diff
@@ -117,18 +110,18 @@ export class CIMaterial extends Component<CIMaterialProps, CIMaterialState> {
...
         return (
             <div className="trigger-modal__trigger">
-                {this.renderIgnoreCache(context)}
+                {this.renderIgnoreCache()}
                 <ButtonWithLoader
                     rootClassName="cta-with-img cta-with-img--ci-trigger-btn"
                     loaderColor="#ffffff"
                     disabled={!canTrigger}
                     isLoading={this.props.isLoading}
                     onClick={(e) => {
```
Comment (vivek-devtron) [inline src/components/app/details/triggerView/ciMaterial.tsx:122]: Extract this inline method

File: src/components/app/details/triggerView/EmptyStateCIMaterial.tsx (hunk 1 above)
Comment (sohel-rp) [reply src/components/app/details/triggerView/EmptyStateCIMaterial.tsx]: Done

File: src/components/app/details/triggerView/MaterialSource.tsx (hunk 2 above)
Comment (sohel-rp) [reply src/components/app/details/triggerView/MaterialSource.tsx]: Done. Also, converted this Class component to FC.

File: src/components/app/details/triggerView/ciMaterial.tsx (hunk 3 above)
Comment (sohel-rp) [reply src/components/app/details/triggerView/ciMaterial.tsx]: Done.

**Comment by sohel-rp** [comment]
@vivek-devtron moved messaging to constants for different components present in this PR.

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**